gradle/build-logic/check-dexer/src/main/kotlin/DexerUtils.kt

```diff
@@ -0,0 +1,128 @@
+import org.gradle.api.GradleException
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.attributes.Attribute
+import org.gradle.api.attributes.Usage
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.JavaExec
+import org.gradle.kotlin.dsl.named
+import org.gradle.kotlin.dsl.register
+import java.io.File
+import java.io.OutputStream
+
+/**
+ * Configures the dexer settings for a given project:
+ * - Adds required repositories
+ * - Sets up r8 and d8 configurations
+ * - Registers the checkDexer task that compiles the project's bytecode with D8
+ *
+ * @param project The project to configure
+ * @param dexerPreCheckTask The prerequisite check task
+ */
+fun configureDexerForProject(project: Project, dexerPreCheckTask: Provider<Task>) {
+    // Setup repositories
+    project.repositories.apply {
+        google()
+        mavenCentral()
+    }
+
+    // Setup r8 configuration
+    if (project.configurations.findByName("r8") == null) {
+        project.configurations.create("r8")
+        project.dependencies.add("r8", "com.android.tools:r8:8.7.18")
+    }
+
+    // Setup d8 configuration
+    configureD8Configuration(project)
+
+    // Register the checkDexer task
+    registerCheckDexerTask(project, dexerPreCheckTask)
+}
+
+/**
+ * Configures the D8 configuration with appropriate attributes.
+ */
+private fun configureD8Configuration(project: Project) {
+    if (project.configurations.findByName("d8") == null) {
+        val d8Configuration = project.configurations.create("d8").apply {
```

Comment:
I think you could write it as follows (same for next apply) or is there a difference?
```suggestion
        val d8Configuration = project.configurations.create("d8") {
```

Reply:
done :)